backend/app/core/booking.py

```
@@ -389,104 +428,123 @@ def _generate_tickets(self, db: Session, booking: Booking) -> List[Ticket]:
             if provider == "entrio":
                 return self._sync_with_entrio(booking, action)
             elif provider == "eventbrite":
                 return self._sync_with_eventbrite(booking, action)
             else:
                 logger.warning(f"Unsupported external provider: {provider}")
                 return {"status": "not_supported", "provider": provider}
-                
+
         except Exception as e:
             logger.error(f"External provider sync error: {e}")
             raise ExternalProviderError(str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)